ext/mbstring/libmbfl/filters/mbfilter_big5.c

```diff
@@ -407,8 +407,12 @@ static size_t mb_big5_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf
 				unsigned int w = (c - 0xA1)*157 + c2 - ((c2 <= 0x7E) ? 0x40 : 0xA1 - 0x3F);
 				ZEND_ASSERT(w < big5_ucs_table_size);
 				w = big5_ucs_table[w];
-				if (!w)
+				if (!w) {
+					if (c == 0xC8) {
+						p--;
+					}
```

Comment:
I suppose that doing this check in its own if branch (even if added ``UNEXPECTED()`` around it) has the same issue as currently?

Reply:
The point of putting it here is that we don't perform the check at all, *unless* an error is detected. If the input string is valid, we just cruise through it without even considering whether the 1st byte of a 2-byte character has this invalid value.

Reply:
ACK